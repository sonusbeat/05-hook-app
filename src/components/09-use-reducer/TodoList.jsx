import TodoItem from "./TodoItem";
import PropTypes from 'prop-types';

const TodoList = ({ HandleToggle, HandleDelete, todos }) => {
  return (
    <table className="table">
      <thead>
        <tr>
          <th>Task</th>
          <th className="text-center">Done</th>
          <th className="text-center">Actions</th>
        </tr>
      </thead>
      <tbody>
        {
          todos.map((todo, index ) => (
            <TodoItem
              key={ todo.id }
              HandleToggle={ HandleToggle }
              HandleDelete={ HandleDelete }
              todo={ todo }
              index={ index }
            />
          ))
        }
      </tbody>
    </table>
  );
};

export default TodoList;

TodoList.propTypes = {
  HandleToggle: PropTypes.func.isRequired,
  HandleDelete: PropTypes.func.isRequired,
  todos: PropTypes.array.isRequired,
};