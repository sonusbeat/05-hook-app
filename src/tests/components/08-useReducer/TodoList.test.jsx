import { shallow } from 'enzyme';
import TodoList from "../../../components/09-use-reducer/TodoList";
import demoTodos from '../../fixtures/demotTodos';

describe('Pruebas en <TodoList />', () => {
  const HandleToggle = jest.fn();
  const HandleDelete = jest.fn();

  const wrapper = shallow(
    <TodoList
      HandleToggle={ HandleToggle }
      HandleDelete={ HandleDelete }
      todos={ demoTodos }
    />
  );

  test('Deberia coincidir con el snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test(`Debe de tener ${demoTodos.length} items`, () => {
    expect(wrapper.find("TodoItem").length).toBe(demoTodos.length);
  });

  test("Su propiedad HandleToggle debe ser enviada como una función", () => {
    expect( wrapper.find("TodoItem").at(0).prop("HandleToggle") )
      .toEqual( expect.any(Function) );
  });

  test("Su propiedad HandleDelete debe ser enviada como una función", () => {
    expect( wrapper.find("TodoItem").at(0).prop("HandleDelete") )
      .toEqual( expect.any(Function) );
  });

});