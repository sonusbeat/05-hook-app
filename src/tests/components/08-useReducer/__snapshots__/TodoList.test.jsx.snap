// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList /> Deberia coincidir con el snapshot 1`] = `
<table
  className="table"
>
  <thead>
    <tr>
      <th>
        Task
      </th>
      <th
        className="text-center"
      >
        Done
      </th>
      <th
        className="text-center"
      >
        Actions
      </th>
    </tr>
  </thead>
  <tbody>
    <TodoItem
      HandleDelete={[MockFunction]}
      HandleToggle={[MockFunction]}
      index={0}
      key="1"
      todo={
        Object {
          "description": "Aprender React",
          "done": false,
          "id": 1,
        }
      }
    />
    <TodoItem
      HandleDelete={[MockFunction]}
      HandleToggle={[MockFunction]}
      index={1}
      key="2"
      todo={
        Object {
          "description": "Aprender ExpressJs",
          "done": false,
          "id": 2,
        }
      }
    />
  </tbody>
</table>
`;
